File: 812-rotate-string/rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
          if (s.length() != goal.length()) {
            return false;
        }

        string my = s;

        for (size_t i = 0; i < my.length(); ++i) {
            if (my == goal) {
                return true;
            }
            rotate(my.begin(), my.begin() + 1, my.end());
        }

        return false;
    }
};

File: 812-rotate-string/rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
         if (s.length() != goal.length()) {
            return false;
        }
        
        String b = s;
        char[] my = b.toCharArray();
        
        for (int i = 0; i < my.length; i++) {
            if (new String(my).equals(goal)) {
                return true;
            }
            char firstChar = my[0];
            System.arraycopy(my, 1, my, 0, my.length - 1);
            my[my.length - 1] = firstChar;
        }
        
        return false;
    }
}